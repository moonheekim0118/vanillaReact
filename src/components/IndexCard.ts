interface Props {
    Quote: string;
    Image: string;
}

const IndexCard = () => {
    function render({ Quote, Image }: Props) {
        const Card = document.createElement("div");
        Card.className = "Index-Card";

        const QuoteContainer = document.createElement("span");
        QuoteContainer.className = "Quote";
        const ImageContainer = document.createElement("img");
        ImageContainer.className = "Image";

        QuoteContainer.innerText = Quote;
        ImageContainer.src = Image;

        Card.appendChild(ImageContainer);
        Card.appendChild(QuoteContainer);

        return Card;
    }

    return render;
};

export default IndexCard;
