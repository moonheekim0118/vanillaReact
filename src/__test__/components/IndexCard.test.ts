import "jsdom-global/register";
import IndexCard from "../../components/IndexCard";

describe("<IndexCard/>", () => {
    let container = null;
    let MockData = {
        Quote: "this is a test",
        Image: "test Image"
    };

    it("renders correctly", () => {
        container = IndexCard(MockData);
    });

    it("matches snapshot", () => {
        expect(container).toMatchSnapshot();
    });
});
