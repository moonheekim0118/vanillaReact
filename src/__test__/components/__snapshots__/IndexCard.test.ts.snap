// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexCard/> matches snapshot 1`] = `
<div
  class="Index-Card"
>
  <span
    class="Card-Title"
  />
  <img
    class="Image"
    src="test Image"
  />
  <span
    class="Quote"
  />
</div>
`;
